homeassistant/components/opnsense/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""Support for OPNSense Routers."""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_URL, CONF_API_KEY, CONF_VERIFY_SSL
+from homeassistant.helpers.discovery import load_platform
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_API_SECRET = "api_secret"
+CONF_TRACKER_INTERFACE = "tracker_interfaces"
+
+DOMAIN = "opnsense"
+
+OPNSENSE_DATA = DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_URL): cv.url,
+                vol.Required(CONF_API_KEY): cv.string,
+                vol.Required(CONF_API_SECRET): cv.string,
+                vol.Optional(CONF_VERIFY_SSL, default=False): cv.boolean,
+                vol.Optional(CONF_TRACKER_INTERFACE, default=[]): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the opnsense component."""
+
+    from pyopnsense import diagnostics
```

Comment:
Please move 3rd party imports to the top of the module.

Reply:
Done in: https://github.com/home-assistant/home-assistant/pull/26834/commits/d1d764b8ef25190603b2c85b43dc4e13d7f93451

But, see the commit message on: https://github.com/home-assistant/home-assistant/pull/26834/commits/cd1bbe60d0140a59fdd547fc35db8b1747afb36a (which made this change) it was the only way I managed to get the mocking working reliably. It only comes up now when I make the change you suggested in your other review comment.